Content.Server/DeviceNetwork/NetworkPayload.cs

```diff
@@ -0,0 +1,50 @@
+using Robust.Shared.Log;
+using System.Collections.Generic;
+using Robust.Shared.Utility;
+using System.Diagnostics.CodeAnalysis;
+
+namespace Content.Server.DeviceNetwork
+{
+    public class NetworkPayload : Dictionary<string, object>
+    {
+        private NetworkPayload(int size) : base(size)
+        {
+        }
+
+        /// <summary>
+        /// Crates a device network packet, drops invalid key value pairs
+        /// </summary>
+        /// <param name="data">An array of tuples containing key value pairs</param>
+        /// <returns>The created device network packet</returns>
+        public static NetworkPayload Create(params (string, object)[] data)
+        {
+            var packet = new NetworkPayload(data.Length);
+            for (var index = 0; index < data.Length; index++)
+            {
+                if (!packet.TryAdd(data[index].Item1, data[index].Item2))
+                {
+                    Logger.Error($"Duplicate device network payload entry: {data[index].Item1}");
+                }
+            }
+            return packet;
+        }
```

Comment:
why not just use the constructor and initialize with C# dictionary syntax?

furthermore, the size field seems unnecessary, you can get the length of a dictionary with the Count property anyway

c# dictionary initializing syntax would make more sense, and you could just remove this strange Create() function and its checks entirely since that's already handled by the dictionary:

```c#
var payload = new NetworkPayload
{
    ["test"] = 1,
    ["test2"] = 2,
    ["test3"] = 3,
}

Reply:
I removed the create function and changed NetworkPayload to just use the dictionary syntax by removing the private constructor